src/request/change-password/ChangePassword.js

```diff
@@ -0,0 +1,211 @@
+            minLength: this._request.keyInfo.hasPin ? 6 : undefined,
+            hideCancel: true,
+            hideInput: !this._request.keyInfo.encrypted,
+        });
+
+        // Adapt to type of secret
+        if (this._request.keyInfo.type === Nimiq.Secret.Type.PRIVATE_KEY) {
+            this._loginFileIcon.setFileUnavailable(true);
+            /** @type {HTMLDivElement} */ (this.$enterPassword.querySelector('.nq-text')).classList.add('display-none');
+            /** @type {HTMLDivElement} */ (this.$setPassword.querySelector('.nq-text')).classList.add('hidden');
+        }
+
+        // Progress Indicators
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.ENTER_PASSWORD} .progress-indicator`),
+            4,
+            1,
+        );
+        this._progressIndicator = new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.SET_PASSWORD} .progress-indicator`),
+            4,
+            2,
+        );
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.DOWNLOAD_FILE} .progress-indicator`),
+            4,
+            4,
+        );
+
+        // Events
+
+        this._passwordGetter.on(PassphraseBox.Events.SUBMIT, this._unlock.bind(this));
+
+        this._passwordSetter.on(PassphraseSetterBox.Events.ENTERED, () => {
+            let colorClass = '';
+            if (this._key.secret instanceof Nimiq.Entropy) {
+                const color = Iqons.getBackgroundColorIndex(
+                    new Nimiq.Address(
+                        // use color of first address as loginFile color
+                        this._key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+                    ).toUserFriendlyAddress(),
+                );
+                const colorString = LoginFile.CONFIG[color].name;
+                colorClass = `nq-${colorString}-bg`;
+            }
+            this._loginFileIcon.lock(colorClass);
+            this._progressIndicator.setStep(3);
+        });
+
+        this._passwordSetter.on(PassphraseSetterBox.Events.SUBMIT, /** @param {string} password */ async password => {
+            const passwordBytes = Utf8Tools.stringToUtf8ByteArray(password);
+
+            await KeyStore.instance.put(this._key, passwordBytes);
+
+            if (this._key.secret instanceof Nimiq.PrivateKey) {
+                this._resolve({ success: true });
+                return;
+            }
+
+            // Prepare Login File for download
+
+            const firstAddress = new Nimiq.Address(
+                this._key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+            );
+
+            const encryptedEntropy = await this._key.secret.exportEncrypted(passwordBytes);
+
+            downloadLoginFile.setEncryptedEntropy(
+                /** @type {Nimiq.SerialBuffer} */ (encryptedEntropy),
```

Comment:
Maybe, instead of double-encryption, we could read the encrypted entropy from the database after it was stored and hand it to the LoginFile from there.

Reply:
How do I do that?